File: week-05/ek_10799_iron_stick.cpp
#include <iostream>
#include <stack>
using namespace std;
int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    string ironStick;
    cin >> ironStick;

    int answer = 0;
    stack<char> st; 
    for(int idx = 0; idx < ironStick.length(); idx++) {
        if (ironStick[idx] == '(')
            st.push(ironStick[idx]);
        else {
            st.pop();
            if (ironStick[idx-1] == '(')
                answer += st.size();
            else
                answer += 1;
        }
    }

    cout << answer;
    return 0;
}

File: week-05/ek_10799_iron_stick.py
import sys
read = sys.stdin.readline

iron_stick = read().rstrip()

st = []
answer = 0
for idx, ch in enumerate(iron_stick):
    if ch == '(':
        st.append(ch)
    else:
        st.pop()
        # 직전이 닫는 괄호면 레이저 아니면 막대기
        answer += (len(st) if iron_stick[idx-1] == '(' else 1)
print(answer)